tests/api_tests/tokens/test_views.py

```diff
@@ -0,0 +1,277 @@
+        bad_text = "<a href='http://sanitized.name'>User_text</a>"
+        cleaned_text = sanitize.strip_html(bad_text)
+
+        payload = copy.copy(self.sample_data)
+        payload['data']['attributes']['name'] = bad_text
+
+        res = self.app.post_json_api(self.user1_list_url, payload, auth=self.user1.auth)
+        assert_equal(res.status_code, 201)
+        assert_equal(res.json['data']['attributes']['name'], cleaned_text)
+
+    def test_created_tokens_show_up_in_api_list(self):
+        res = self.app.post_json_api(self.user1_list_url, self.sample_data, auth=self.user1.auth)
+        assert_equal(res.status_code, 201)
+
+        res = self.app.get(self.user1_list_url, auth=self.user1.auth)
+        assert_equal(len(res.json['data']),
+                     len(self.user1_tokens) + 1)
+
+    def test_returns_401_when_not_logged_in(self):
+        res = self.app.get(self.user1_list_url, expect_errors=True)
+        assert_equal(res.status_code, 401)
+
+    def tearDown(self):
+        super(TestTokenList, self).tearDown()
+        ApiOAuth2PersonalToken.remove()
+        User.remove()
+
+
+class TestTokenDetail(ApiTestCase):
+    def setUp(self):
+        super(TestTokenDetail, self).setUp()
+
+        self.user1 = AuthUserFactory()
+        self.user2 = AuthUserFactory()
+
+        self.user1_token = ApiOAuth2PersonalTokenFactory(owner=self.user1)
+        self.user1_token_url = _get_token_detail_route(self.user1_token)
+
+        self.missing_type = {
+            'data': {
+                'attributes': {
+                    'name': 'A shiny updated token',
+                    'scopes': ['osf.full_write'],
+                }
+            }
+        }
+
+        self.incorrect_type = {
+            'data': {
+                'type': 'Wrong type.',
+                'attributes': {
+                    'name': 'A shiny updated token',
+                    'scopes': ['osf.full_write'],
+                }
+            }
+        }
+
+        self.correct = {
+            'data': {
+                'type': 'tokens',
+                'attributes': {
+                    'name': 'A shiny updated token',
+                    'scopes': ['osf.full_write'],
+                }
+            }
+        }
+
+    def test_owner_can_view(self):
+        res = self.app.get(self.user1_token_url, auth=self.user1.auth)
+        assert_equal(res.status_code, 200)
+        assert_equal(res.json['data']['id'], self.user1_token._id)
+
+    def test_non_owner_cant_view(self):
+        res = self.app.get(self.user1_token_url, auth=self.user2.auth, expect_errors=True)
+        assert_equal(res.status_code, 403)
+
+    def test_returns_401_when_not_logged_in(self):
+        res = self.app.get(self.user1_token_url, expect_errors=True)
+        assert_equal(res.status_code, 401)
+
+    # @mock.patch('framework.auth.cas.CasClient.revoke_personal_tokens')
+    # def test_owner_can_delete(self, mock_method):
```

Comment:
There are multiple places in the tests where code is simply commented out, and I'm not sure why. Are you keping them in for discussion? Otherwise this should really have been reviewed and cleaned up before advancing to PR. 

Please review and make sure that the tests are appropriately selected for this task, rather than just a modified copy-paste.